KerbalStuff/config.py

```diff
@@ -15,13 +15,27 @@ def get_env_var_or_config(section, key):
     if env_var:
         return env_var
     else:
-        return config.get(section, key)
+        return config.get(section, key, fallback=None)
 
 
-_cfg = lambda k: get_env_var_or_config(env, k)
-_cfgi = lambda k: int(_cfg(k))
-_cfgb = lambda k: strtobool(_cfg(k)) == 1
-_cfgl = lambda k: ast.literal_eval(_cfg(k))
+def _cfg(k):
+    return get_env_var_or_config(env, k)
+
+
+def _cfgi(k):
+    val = _cfg(k)
+    return int(val) if val is not None else 0
+
+
+def _cfgb(k):
+    val = _cfg(k)
+    return strtobool(val) == 1 if val is not None else False
+
+
+def _cfgl(k):
+    val = _cfg(k)
+    return ast.literal_eval(val) if val is not None else {}
```

Comment:
Even stronger here: `ast.literal_eval` may produce **any** python native type, not only `dict`
```suggestion
    return ast.literal_eval(val) if val is not None else None
```

Reply:
But what will we be using it for? For strings we have `_cfg`, for ints we have `_cfgi`, for bools we have `_cfgb`. This function is for associative arrays; if we treat it that way, then we don't need extra code to keep such usages from crashing.

https://github.com/KSP-SpaceDock/SpaceDock/blob/413177db00b995630303e98713dedbbde7bfa9a9/templates/layout.html#L185

Reply:
This function is _not_ for dicts because nothing in its name (let alone missing docstring) suggests this; not dees it returns only dicts -- it would return ints, floats, strings, **lists**, **tuples** (which are more common in configs than dicts), dicts and so on. There's absolutely no basis for this function to be restricted to dicts and, of all things, to return an empty _dict_ as default.

So in this case I insist on default being `None`.

Reply:
OK, renamed it from `_cfgl` signifying "literal" to `_cfgd` signifying "dictionary." This was easy since it's only used in one place.